File: src/main/scala/nl/rug/jbi/search/ssap/util/PathSuffixFilter.scala
package nl.rug.jbi.search.ssap.util

import java.io.File
import org.apache.commons.io.filefilter.AbstractFileFilter

class PathSuffixFilter(suffix: String) extends AbstractFileFilter {
  override def accept(file: File): Boolean = file.getCanonicalPath.endsWith(suffix)
}

File: src/main/java/nl/rug/jbi/search/ssap/util/PathSuffixFilter.java
package nl.rug.jbi.search.ssap.util;

import java.io.File;
import java.io.IOException;
import org.apache.commons.io.filefilter.AbstractFileFilter;

public class PathSuffixFilter extends AbstractFileFilter {
    public String suffix;

    public PathSuffixFilter(final String suffix) {
        this.suffix = suffix;
    }

    public boolean accept(File file) {
        try {
            return file.getCanonicalPath().endsWith(suffix);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}
